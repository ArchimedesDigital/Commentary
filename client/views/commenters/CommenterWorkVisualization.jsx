
CommenterWorkVisualization = React.createClass({

	propTypes: {
		work: React.PropTypes.object.isRequired,
		toggleVisibleWork: React.PropTypes.func.isRequired,
		commenterWordpressId: React.PropTypes.number.isRequired,
	},

	toggleVisibleWork(workSlug) {
		this.props.toggleVisibleWork(workSlug);
		this.refs.workVisualization.close();
	},

	render() {
		const work = this.props.work;

		return (
			<div className={`commenter-work-visualization commenter-work-visualization--${work.slug}`}>
				<i
					onClick={this.toggleVisibleWork.bind(null, work.slug)}
					className="close-visualization mdi mdi-close"
				/>
				<WorkVisualization
					ref='workVisualization'
					work={work}
					commenterWordpressId={this.props.commenterWordpressId}
				/>
			</div>
		);
	},
});
