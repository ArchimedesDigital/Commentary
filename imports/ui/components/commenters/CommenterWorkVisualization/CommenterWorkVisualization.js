import React from 'react';

import WorkVisualization from '/imports/ui/components/works/WorkVisualization';


const CommenterWorkVisualization = React.createClass({

	propTypes: {
		work: React.PropTypes.object.isRequired,
		toggleVisibleWork: React.PropTypes.func.isRequired,
		commenterSlug: React.PropTypes.string.isRequired,
		isTest: React.PropTypes.bool,
	},

	toggleVisibleWork(workSlug) {
		this.props.toggleVisibleWork(workSlug);
		this.workVisualization.close();
	},

	render() {
		const { work, commenterSlug, isTest } = this.props;

		return (
			<div className={`commenter-work-visualization commenter-work-visualization--${work.slug}`}>
				<i
					onClick={this.toggleVisibleWork.bind(null, work.slug)}
					className="close-visualization mdi mdi-close"
				/>
				{!isTest ?
					<WorkVisualization
						ref={(component) => { this.workVisualization = component; }}
						work={work}
						commenterSlug={commenterSlug}
					/>
				: ''}
			</div>
		);
	},
});


export default CommenterWorkVisualization;
