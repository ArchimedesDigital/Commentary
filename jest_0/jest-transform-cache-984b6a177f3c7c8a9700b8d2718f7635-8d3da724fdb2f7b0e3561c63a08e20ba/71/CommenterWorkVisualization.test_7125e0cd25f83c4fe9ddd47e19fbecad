/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommenterWorkVisualization = require('./CommenterWorkVisualization'); /* istanbul ignore next */var _CommenterWorkVisualization2 = _interopRequireDefault(_CommenterWorkVisualization);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommenterWorkVisualization', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommenterWorkVisualization2.default, /* istanbul ignore next */{
			work: {
				title: 'Iliad',
				slug: 'iliad',
				subworks: [{
					title: '1',
					slug: '1',
					n: 1 }] },


			commenterSlug: 'test',
			toggleVisibleWork: function /* istanbul ignore next */toggleVisibleWork() {},
			isTest: /* istanbul ignore next */true })).


		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: