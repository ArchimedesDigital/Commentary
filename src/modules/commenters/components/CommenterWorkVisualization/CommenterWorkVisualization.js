import React, { Component } from 'react';
import PropTypes from 'prop-types';

import WorkVisualization from '../../../works/components/WorkVisualization/WorkVizualization';


class CommenterWorkVisualization extends Component {

	constructor(props) {
		super(props);


		this.toggleVisibleWork = this.toggleVisibleWork.bind(this);
	}

	toggleVisibleWork(workSlug) {
		this.props.toggleVisibleWork(workSlug);
		this.workVisualization.close();
	}

	render() {
		const { work, commenterSlug, isTest } = this.props;

		return (
			<div className={`commenter-work-visualization commenter-work-visualization--${work.slug}`}>
				<i
					onClick={this.toggleVisibleWork}
					className="close-visualization mdi mdi-close"
				/>
				{!isTest ?
					<WorkVisualization
						ref={(component) => { this.workVisualization = component; }}
						work={work}
						commenterSlug={commenterSlug}
					/>
					: ''}
			</div>
		);
	}
}

CommenterWorkVisualization.propTypes = {
	work: PropTypes.object.isRequired,
	toggleVisibleWork: PropTypes.func.isRequired,
	commenterSlug: PropTypes.string.isRequired,
	isTest: PropTypes.bool,
};

export default CommenterWorkVisualization;
