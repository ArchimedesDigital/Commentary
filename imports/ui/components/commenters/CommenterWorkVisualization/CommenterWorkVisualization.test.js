import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommenterWorkVisualization from './CommenterWorkVisualization';

describe('CommenterWorkVisualization', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<CommenterWorkVisualization
					work={{
						title: 'Iliad',
						slug: 'iliad',
						subworks: [{
							title: '1',
							slug: '1',
							n: 1,
						}],
					}}
					commenterSlug="test"
					toggleVisibleWork={() => {}}
					isTest
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
