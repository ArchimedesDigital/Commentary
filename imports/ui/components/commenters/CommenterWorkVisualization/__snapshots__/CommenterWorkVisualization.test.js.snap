// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommenterWorkVisualization renders correctly 1`] = `
<div
  className="commenter-work-visualization commenter-work-visualization--iliad"
>
  <i
    className="close-visualization mdi mdi-close"
    onClick={[Function]}
  />
  
</div>
`;
